docs/modules/concepts/pages/storage/index.adoc

```diff
@@ -0,0 +1,112 @@
+    locate a message that you may want to refer to again in the future
+ * As an archive &mdash; a place where you may want to store "deleted" emails so that
+    you can potentially retrieve them if you wish to change your mind later
+ * As a temporary step in a process &mdash; perhaps you have a "todo" folder where
+    you temporarily store an email that you intend to process soon
+
+There are numerous ways to store emails, just as there are numerous reasons for
+storing them:
+
+ * A <<Specifications,standardized>> "mailbox" format
+ * A relational database
+ * A custom file-based mechanism
+ * Etc.
+
+This section briefly describes how email storage is a core concept in an
+email system.
+
+
+
+== Specifications
+
+=== Electronic Mail Box
+
+==== RFC196
+
+The concept of an "electronic mail box" was 
+https://tools.ietf.org/html/rfc196[initially specified] way back in 1971.
+It is amusing to read the retro concepts in the initial description of this
+document:
+
+  The purpose of this protocol is to provide at each site a
+  standard mechanism to receive sequential files for immediate or
+  deferred printing or other uses.  The files for deferred printing
+  would probably be stored on intermediate disk files [...].
+  It is also assumed that there would be a program at the sending
+  site [...]. This program could probably be accessed as a subcommand 
+  of the Telnet program.
+
+This specification was obsoleted by <<RFC221>>.
+
+
+==== RFC221
+
+https://tools.ietf.org/html/rfc221[RFC221] ("A Mail Box Protocol, Version-2") was
+published in 1971. It added the possibility of using FTP in addition to the
+"Data Transfer Protocol" used in <<RFC196>>. This version was obsoleted by
+<<RFC278>>.
+
+
+
+==== RFC278
+
+https://tools.ietf.org/html/rfc278[RFC278] ("Revision of the Mail Box Protocol")
+was published in 1971. It provided a number of updates to <<RFC221>>.
+
+
+
+=== Maildir
+
+https://en.wikipedia.org/wiki/Maildir
+
+=== Mbox
+
+https://en.wikipedia.org/wiki/Mbox
+
+
+==== RFC4155
+
+https://tools.ietf.org/html/rfc4155[RFC4155] ("The application/mbox Media Type").
+
+=== Note
+
+RFC 5322 defines a mailbox as follows:[2] A mailbox receives mail. It is a 'conceptual entity' that does not necessarily pertain to file storage. It further exemplifies that some sites may choose to print mail on a printer and deliver the output to the addressee's desk, much like a traditional fax transmission.
+
+
+== Related Concepts
+
+Email Storage is related to...
+
+ * *xref:protocols/pop.adoc[POP]*, as the POP protocol mandates interaction
+   with a user's "mailbox"
+ * *xref:protocols/imap.adoc[POP]*, as the IMAP protocol is all about storing
+   messages on an IMAP Server
+ * *Email clients*, as the client will store mail locally, usually in the form
+   of a "mailbox"
+ * *xref:protocols/smtp.adoc[SMTP]* as the protocol is related to transmitting
+   messages from one mailbox to another
+
+
+== James Model
+
+(TODO)
```

Comment:
Should this speak of the mailbox-api, & the various implementation we have of it?

I would be happy to write such a section.

Reply:
That would be really nice if you could give it a go!

Reply:
I will add this information to a different section, and keep this section reserved for more general concepts.